cmd/geth/js.go

```diff
@@ -425,12 +437,13 @@ func (self *jsre) interactive() {
 	}
 }
 
-func hidepassword(input string) string {
-	if passwordRegexp.MatchString(input) {
-		return passwordRepl
-	} else {
-		return input
+func postprocess(input string) (string, bool) {
```

Comment:
I think @ethers (https://github.com/ethereum/go-ethereum/issues/1697) asked for this feature not to hide passwords but to prevent dangerous commands not to end up in the history from which they can accidentally be executed again when the users browses through history. That means that besides password interaction also transaction confirmation from the console should be enabled since transactions will have irreversible consequences.